examples/src/bin/simpleserver.rs

```diff
@@ -0,0 +1,51 @@
+//! This is the simplest possible server using rustls that does something useful:
+//! it accepts the default configuration, loads a server certificate and private key,
+//! and then accepts a single client connection.
+//!
+//! You must either set the CERTFILE and PRIV_KEY_FILE env vars to point to a server
+//! certificate and private key, or place 'localhost.pem' and 'localhost-key.pem' in
+//! the directory you run this example from.
+//!
+//! Note that `unwrap()` is used to deal with networking errors; this is not something
+//! that is sensible outside of example code.
+
+use std::error::Error as StdError;
+use std::fs::File;
+use std::io::{BufReader, Read, Write};
+use std::net::TcpListener;
+use std::sync::Arc;
+
+const CERTFILE: &str = match option_env!("CERTFILE") {
+    Some(certfile) => certfile,
+    None => "localhost.pem",
+};
+
+const PRIV_KEY_FILE: &str = match option_env!("PRIV_KEY_FILE") {
+    Some(priv_key_file) => priv_key_file,
+    None => "localhost-key.pem",
+};
```

Comment:
I was surprised by these, both that they precede `main()` (which seems counter to our usual item ordering) and also by the use of environment variables, whereas we use command-line arguments in what I thought were most of the other examples.

(Even more nitty, I also find the naming of `CERT_FILE` without underscore and the `PRIV` abbreviation instead of `PRIVATE_` too be not quite idiomatic.)

Reply:
I lifted this from the unbuffered examples that were added recently, so they would need adjustment for this feedback as well

Reply:
https://github.com/rustls/rustls/pull/1719